### DISCUSSION TITLE
TFLINT fails with `Failed to check ruleset`

### DISCUSSION THREAD

**Discussion by pranav-shankar-swi** [terraform-linters/tflint#1617] [Q&A] [answered]
Getting below error while running tflint via MegaLinter v5 (`oxsecurity/megalinter/flavors/terraform@v5`)

Module - 

`main.tf`
```
resource "aws_elasticache_subnet_group" "this" {
  count      = var.create_subnet_group && length(var.subnet_group_name) > 0 ? 1 : 0
  name       = var.subnet_group_name
  subnet_ids = var.subnet_ids
}

resource "aws_elasticache_cluster" "this" {
  cluster_id           = var.redis-server-name
  engine               = "redis"
  node_type            = var.node_type
  num_cache_nodes      = 1
  parameter_group_name = "default.redis3.2"
  engine_version       = "3.2.6"
  port                 = 6379
  security_group_ids   = [var.security_group_id]
  subnet_group_name    = aws_elasticache_subnet_group.this[0].name
}

```

variables.tf
```
variable "redis-server-name" {
  default = ""
}

variable "subnet_ids" {
  type    = list(any)
  default = []
}

variable "security_group_id" {
  type    = string
  default = ""
}

variable "create_subnet_group" {
  type    = bool
  default = true
}

variable "subnet_group_name" {
  type    = string
  default = ""
}

variable "node_type" {
  type    = string
  default = ""
}
```

Error in linter logs --> 

```[tflint] modules/aws/redis/main.tf - ERROR - 1 error(s)
--Error detail:
Failed to check ruleset; error reading from server: EOF

[tflint] modules/aws/redis/variables.tf - ERROR - 1 error(s)
--Error detail:
Failed to check ruleset; error reading from server: EOF```

Can someone please help with this ?

**Comment by bendrucker** [comment]
If you can isolate this to a reproducible error, go ahead and file a [bug report](https://github.com/terraform-linters/tflint/issues/new?assignees=&labels=bug&template=bug.yml). Make sure to follow the instructions in the form as we can only investigate issues that you've isolated from your unique environment, including wrappers like megalinter.

**Comment by pranav-shankar-swi** [comment]
After defining "default" value for node_type in `variables.tf` , we are not seeing this issue anymore

```
 variable "node_type" {
   type    = string
-  default = ""
+  default = "cache.m4.large"
 }
```

**Comment by bendrucker** [comment, marked as answer]
Reproducible report:

https://github.com/terraform-linters/tflint-ruleset-aws/issues/429

While `node_type = ""` is invalid and should emit an issue, the plugin shouldn't crash.

Valid choices include:

* Set an actual `default` node type
* Set the `default` to `null`, implying `replication_group_id` must be provided if `node_type` is unset
* Omit the `default`

**Comment by nvuillam** [comment]
You should upgrade to MegaLinter v6 to have a newer tflint version :)
